src/contracts/Catalyst.ts

```diff
@@ -0,0 +1,103 @@
+  script: string
+  input: string
+  returnData: string
+}
+export type RecoverToVaultEvent = {
+  vault: Address
+  token: Address
+  amount: string
+}
+export type AddCatalystEventLog = EventLog<AddCatalystEvent, 'AddCatalyst'>
+export type RemoveCatalystEventLog = EventLog<RemoveCatalystEvent, 'RemoveCatalyst'>
+export type ScriptResultEventLog = EventLog<ScriptResultEvent, 'ScriptResult'>
+export type RecoverToVaultEventLog = EventLog<RecoverToVaultEvent, 'RecoverToVault'>
+interface CatalystEvents {
+  AddCatalyst: EventSubscriptionFactory<AddCatalystEventLog>
+  RemoveCatalyst: EventSubscriptionFactory<RemoveCatalystEventLog>
+  ScriptResult: EventSubscriptionFactory<ScriptResultEventLog>
+  RecoverToVault: EventSubscriptionFactory<RecoverToVaultEventLog>
+}
+interface CatalystEventLogs {
+  AddCatalyst: AddCatalystEventLog
+  RemoveCatalyst: RemoveCatalystEventLog
+  ScriptResult: ScriptResultEventLog
+  RecoverToVault: RecoverToVaultEventLog
+}
+interface CatalystTxEventLogs {
+  AddCatalyst: AddCatalystEventLog[]
+  RemoveCatalyst: RemoveCatalystEventLog[]
+  ScriptResult: ScriptResultEventLog[]
+  RecoverToVault: RecoverToVaultEventLog[]
+}
+export type CatalystTransactionReceipt = TransactionReceipt<CatalystTxEventLogs>
+interface CatalystMethods {
+  owners(a0: Address): TxCall<boolean>
+  hasInitialized(): TxCall<boolean>
+  catalystCount(): TxCall<string>
+  getEVMScriptExecutor(_script: string): TxCall<Address>
+  getRecoveryVault(): TxCall<Address>
+  catalystIndexById(a0: string): TxCall<string>
+  catalystOwner(_id: string): TxCall<Address>
+  catalystDomain(_id: string): TxCall<string>
+  catalystIds(a0: number | string | BN): TxCall<string>
+  allowRecoverability(token: Address): TxCall<boolean>
+  appId(): TxCall<string>
+  initialize(): TxSend<CatalystTransactionReceipt>
+  getInitializationBlock(): TxCall<string>
+  transferToVault(_token: Address): TxSend<CatalystTransactionReceipt>
+  canPerform(_sender: Address, _role: string, _params: (number | string | BN)[]): TxCall<boolean>
+  getEVMScriptRegistry(): TxCall<Address>
+  removeCatalyst(_id: string): TxSend<CatalystTransactionReceipt>
+  domains(a0: string): TxCall<boolean>
+  catalystById(
+    a0: string
+  ): TxCall<{
+    id: string
+    0: string
+    owner: Address
+    1: Address
+    domain: string
+    2: string
+    startTime: string
+    3: string
+    endTime: string
+    4: string
+  }>
+  addCatalyst(_owner: Address, _domain: string): TxSend<CatalystTransactionReceipt>
+  kernel(): TxCall<Address>
+  MODIFY_ROLE(): TxCall<string>
+  isPetrified(): TxCall<boolean>
+}
+export interface CatalystDefinition {
+  methods: CatalystMethods
+  events: CatalystEvents
+  eventLogs: CatalystEventLogs
+}
+export class Catalyst extends Contract<CatalystDefinition> {
+  constructor(eth: Eth, address?: Address, options?: ContractOptions) {
+    super(eth, abi, address, options)
+  }
+  deploy(): TxSend<CatalystTransactionReceipt> {
+    return super.deployBytecode(
```

Comment:
is this deploy() needed? can it be a noop fn?

Reply:
The code in `contracts/*` is not exposed for clients of this library. I would rather leave the auto-generated code as it is, because if we decide to use it in the future, it will be almost imposible to figure out where the problem is